ingestion-beam/src/test/java/com/mozilla/telemetry/SinkAvroTest.java

```diff
@@ -0,0 +1,119 @@
+  @Test
+  public void testSingleDocumentType() {
+    String input = Resources.getResource("testdata/avro-message-single-doctype.ndjson").getPath();
+    String schemas = Resources.getResource("testdata/avro-schema-test.tar.gz").getPath();
+    String output = outputPath + "/out";
+
+    Sink.main(new String[] { "--inputFileFormat=json", "--inputType=file", "--input=" + input,
+        "--outputType=avro", "--output=" + output, "--outputFileCompression=UNCOMPRESSED",
+        "--schemaLocation=" + schemas, "--errorOutputFileCompression=UNCOMPRESSED",
+        "--errorOutputType=stdout" });
+
+    assertThat("output count", getPrefixFileCount(outputPath, "out"), Matchers.greaterThan(0L));
+  }
+
+  /**
+   * Test that documents with existing schemas are being written out to the correct location.
+   */
+  @Test
+  public void testMultipleDocumentTypes() {
+    String input = Resources.getResource("testdata/avro-message-multiple-doctype.ndjson").getPath();
+    String schemas = Resources.getResource("testdata/avro-schema-test.tar.gz").getPath();
+    String output = outputPath + "/${document_type:-default}/out";
+
+    Sink.main(new String[] { "--inputFileFormat=json", "--inputType=file", "--input=" + input,
+        "--outputType=avro", "--output=" + output, "--outputFileCompression=UNCOMPRESSED",
+        "--schemaLocation=" + schemas, "--errorOutputFileCompression=UNCOMPRESSED",
+        "--errorOutputType=stdout" });
+
+    assertThat("foo output count", getPrefixFileCount(outputPath + "/foo", "out"),
+        Matchers.greaterThan(0L));
+    assertThat("bar output count", getPrefixFileCount(outputPath + "/bar", "out"),
+        Matchers.greaterThan(0L));
+    assertThat("baz output count", getPrefixFileCount(outputPath + "/baz", "out"),
+        Matchers.greaterThan(0L));
+
+  }
+
+  /**
+   * Test that for invalid documents. This case includes a ping where the data
+   * wouldn't have passed validation and a document where a schema doesn't
+   * exist. These documents should go to the error collection so they can be
+   * reprocessed.
+   */
+  @Test
+  public void testInvalidDocuments() {
+    String input = Resources.getResource("testdata/avro-message-invalid-doctype.ndjson").getPath();
+    String schemas = Resources.getResource("testdata/avro-schema-test.tar.gz").getPath();
+    String output = outputPath + "/${document_type:-default}/out";
+    String errorOutput = outputPath + "/err";
+
+    Sink.main(new String[] { "--inputFileFormat=json", "--inputType=file", "--input=" + input,
+        "--outputType=avro", "--output=" + output, "--outputFileCompression=UNCOMPRESSED",
+        "--schemaLocation=" + schemas, "--errorOutputFileCompression=UNCOMPRESSED",
+        "--errorOutputType=file", "--errorOutput=" + errorOutput, });
+
+    assertThat("error count", getPrefixFileCount(outputPath, "err"), Matchers.greaterThan(0L));
+    assertThat("foo output count", getPrefixFileCount(outputPath + "/foo", "out"),
+        Matchers.is(-1L));
+  }
```

Comment:
Add a test asserting the correctness of a record. Additionally, a test verifying the error output is including the correct exceptions. These tests were written before anything worked, so these tests are based on the expected file outputs.